app/controllers/api/functions.php

```diff
@@ -1568,3 +1362,165 @@
+    ->param('range', '30d', new WhiteList(['24h', '7d', '30d', '90d']), 'Date range.', true)
+    ->inject('response')
+    ->inject('dbForProject')
+    ->action(function (string $functionId, string $range, Response $response, Database $dbForProject) {
+
+        $function = $dbForProject->getDocument('functions', $functionId);
+
+        if ($function->isEmpty()) {
+            throw new Exception(Exception::FUNCTION_NOT_FOUND);
+        }
+
+        $periods = Config::getParam('usage', []);
+        $stats = $usage = [];
+        $days = $periods[$range];
+        $metrics = [
+            'functions.' . $function->getInternalId() . '.deployments',
+            'functions.' . $function->getInternalId() . '.deployments.storage',
+            $function->getInternalId() . '.builds',
+            $function->getInternalId() . '.builds.storage',
+            $function->getInternalId() . '.builds.compute',
+            $function->getInternalId() . '.executions',
+            $function->getInternalId() . '.executions.compute',
+        ];
+
+        Authorization::skip(function () use ($dbForProject, $days, $metrics, &$stats) {
+            foreach ($metrics as $metric) {
+                $limit = $days['limit'];
+                $period = $days['period'];
+                $results = $dbForProject->find('stats', [
+                    Query::equal('period', [$period]),
+                    Query::equal('metric', [$metric]),
+                    Query::limit($limit),
+                    Query::orderDesc('time'),
+                ]);
+                $stats[$metric] = [];
+                foreach ($results as $result) {
+                    $stats[$metric][$result->getAttribute('time')] = [
+                        'value' => $result->getAttribute('value'),
+                    ];
+                }
+            }
+        });
+
+        $format = match ($days['period']) {
+            '1h' => 'Y-m-d\TH:00:00.000P',
+            '1d' => 'Y-m-d\T00:00:00.000P',
+        };
+
+    foreach ($metrics as $metric) {
+        $usage[$metric] = [];
+        $leap = time() - ($days['limit'] * $days['factor']);
+        while ($leap < time()) {
+            $leap += $days['factor'];
+            $formatDate = date($format, $leap);
+            $usage[$metric][] = [
+                'value' => $stats[$metric][$formatDate]['value'] ?? 0,
+                'date' => $formatDate,
+            ];
+        }
+    }
+
+        $response->dynamic(new Document([
+            'range' => $range,
+            'deployments' => $usage[$metrics[0]],
+            'deploymentsStorage' => $usage[$metrics[1]],
+            'builds' => $usage[$metrics[2]],
+            'buildsStorage' => $usage[$metrics[3]],
+            'buildsCompute' => $usage[$metrics[4]],
+            'executions' => $usage[$metrics[5]],
+            'executionsCompute' => $usage[$metrics[6]],
+        ]), Response::MODEL_USAGE_FUNCTION);
+    });
+
+App::get('/v1/functions/usage')
```

Comment:
Can you please keep this endpoint back on Line 321 ? Else this diff is hard to interpret .

Reply:
ok